pmm-app/src/pmm-qan-app-panel/QueryAnalytics/StateContext.tsx

```diff
@@ -0,0 +1,279 @@
+    return FILTERS_NAMES.reduce((result, filterName) => {
+      const filters = query.getAll(`var-${filterName}`);
+      if (!filters.length) {
+        return result;
+      }
+
+      if (filters[0] === 'All' || filters[0] === '') {
+        return result;
+      }
+
+      result[filterName] = filters;
+      return result;
+    }, {});
+  }
+
+  static refreshGrafanaVariables(state) {
+    const dataSource = getDataSourceSrv();
+    // @ts-ignore
+    const templateVariables = dataSource.templateSrv.variables;
+    FILTERS_NAMES.forEach(filter => {
+      const variables = _.find(templateVariables, { name: filter.replace('var-', '') });
+      if (!variables) {
+        return;
+      }
+      variables.current = {
+        text: state.labels[filter] || ['All'],
+        value: state.labels[filter] || ['All'],
+      };
+    });
+    templateVariables[0].variableSrv.variableUpdated(templateVariables[0]);
+  }
+
+  static generateURL(state) {
+    // read parameters and create new url
+    // @ts-ignore
+    const labels =
+      state.labels &&
+      Object.keys(state.labels)
+        .map(key => {
+          // @ts-ignore
+          const variables = state.labels[key];
+          return variables.map(variable => `var-${key}=${variable}`).join('&');
+        })
+        .filter(Boolean)
+        .join('&');
+    const columnsQuery = state.columns ? `columns=${JSON.stringify(state.columns)}` : '';
+    const groupBy = state.groupBy ? `group_by=${state.groupBy}` : '';
+    const filterByQuery = state.queryId ? `filter_by=${state.queryId}` : '';
+    const orderBy = state.orderBy ? `order_by=${state.orderBy}` : '';
+    const from = state.rawTime && state.rawTime.from ? `from=${state.rawTime.from}` : '';
+    const to = state.rawTime && state.rawTime.to ? `to=${state.rawTime.to}` : '';
+    // TODO: replace crutch with right redirect
+    return `${window.location.pathname}?${[columnsQuery, filterByQuery, labels, orderBy, groupBy, from, to]
+      .filter(Boolean)
+      .join('&')}`;
+  }
+
+  static parseURL(query) {
+    const urlParams = {} as any;
+    urlParams.from = ParseQueryParamDate.transform(query.get('from') || 'now-12h', 'from')
+      .utc()
+      .format('YYYY-MM-DDTHH:mm:ssZ');
+    urlParams.to = ParseQueryParamDate.transform(query.get('to') || 'now', 'to')
+      .utc()
+      .format('YYYY-MM-DDTHH:mm:ssZ');
+    // urlParams.rawTime = {
+    //   from: query.get('from'),
+    //   to: query.get('to'),
+    // };
+    urlParams.columns = JSON.parse(query.get('columns')) || DEFAULT_COLUMNS;
+    urlParams.labels = ContextActions.setFilters(query);
+    urlParams.pageNumber = 1;
+    urlParams.pageSize = 10;
+    urlParams.orderBy = query.get('order_by') || `-${urlParams.columns[0]}`;
+    urlParams.queryId = query.get('filter_by');
+    urlParams.querySelected = !!query.get('filter_by');
+    urlParams.groupBy = query.get('group_by') || 'queryid';
+    // console.log(Object.assign({}, urlParams));
+    return urlParams;
```

Comment:
Consider return an object straight from `return`: it avoids repeating `urlParams` over and over again and reads much better as a result.